pkg/ottl/expression.go

```diff
@@ -117,6 +122,52 @@ func (g StandardTypeGetter[K, T]) Get(ctx context.Context, tCtx K) (T, error) {
 	return v, nil
 }
 
+type StringLikeGetter[K any] interface {
```

Comment:
I think it would be good to have a Godoc comment to explain the intent of this type.

Reply:
Done